jaxrs/src/test/scala/com/cerner/beadledom/jaxrs/exceptionmappers/ThrowableExceptionMapperSpec.scala

```diff
@@ -0,0 +1,138 @@
+package com.cerner.beadledom.jaxrs.exceptionmappers
+
+import com.cerner.beadledom.jaxrs.models.JsonError
+import com.cerner.beadledom.jaxrs.provider.{FakeDao, FakeResource}
+
+import com.fasterxml.jackson.jaxrs.json.JacksonJsonProvider
+
+import org.jboss.resteasy.mock._
+import org.mockito.Mockito
+import org.mockito.Mockito._
+
+import play.api.libs.json.Json
+
+import javax.ws.rs.core.HttpHeaders.CONTENT_TYPE
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.Response.Status._
+
+/**
+  * Unit tests for {@link ThrowableExceptionMapper}.
+  *
+  * @author Cal Fisher
+  */
+class ThrowableExceptionMapperSpec extends BaseExceptionMapperSpec {
+
+  val throwableExceptionMapper = new ThrowableExceptionMapper
+  val fakeDao = mock[FakeDao]
+  val fakeResource = new FakeResource(fakeDao)
+  val url = "/fakeResource/ExceptionEndpoint"
+  val request = MockHttpRequest.get(url)
+  var response: MockHttpResponse = _
+  val internalServerErrorJsonError = createJsonError(INTERNAL_SERVER_ERROR)
+
+  override def beforeEach {
+    Mockito.reset(fakeDao)
+    response = new MockHttpResponse
+  }
+
+  describe("ThrowableExceptionMapper") {
+    describe("unit testing the behavior of ThrowableExceptionMapper class when an exception is " +
+        "thrown from a resource") {
+      val dispatcher = MockDispatcherFactory.createDispatcher()
+      dispatcher.getRegistry.addSingletonResource(fakeResource)
+      dispatcher.getProviderFactory.registerProvider(classOf[JacksonJsonProvider])
+      dispatcher.getProviderFactory.registerProvider(classOf[ThrowableExceptionMapper])
```

Comment:
I think you could create the `dispatcher` above with the other `val`s and configure it in a `beforeAll`. Then you could remove the describe that has the really long sentence.

Reply:
While writing tests, I tend to split them up with `describe` blocks when different types of functionality are happening, so the `describe` block with the really long sentence is meant to indicate that an actual resource is being used. Further down in the spec, you'll see another `describe` block for simply unit testing the class directly. Any suggestions for better naming?

I can move the dispatcher up with the other `val`s like you're suggesting and configure it in a `beforeAll` block, but that would require extending `BeforeAndAfterAll`, which also has the cumbersome syntax: 
```
override def beforeAll(): Unit = {
  ...
}
```

I couldn't find any simpler syntax alternatives (relating back to your suggestion with `BeforeAndAfter` vs `BeforeAndAfterEach`). Would you still prefer I made this change with the `beforeAll`?

Reply:
Nesting setup in `describe` blocks can result in weird behavior. All code within `describe` blocks, but not within an `it` block will be executed when the spec is being initialized rather than when the nested tests are actually run.

Reply:
I removed the setup of the dispatcher from the `describe` block to above with the `val`s and inside of a `beforeAll` (for all three specs): https://github.com/cerner/beadledom/pull/45/commits/e1721a2548f63bd69d56eba9dcd838de312ed2d7#diff-0734403a349f9724fd379e4bead2c7e8R32